src/client/lazy-app/Modal/index.tsx

```diff
@@ -0,0 +1,136 @@
+import { h, Component, VNode, Fragment } from 'preact';
+import * as style from './style.css';
+import 'add-css:./style.css';
+import { linkRef } from 'shared/prerendered-app/util';
+import { cleanSet } from '../util/clean-modify';
+import { animateTo } from '../util';
+
+interface Props {
+  icon: VNode;
+  title: string;
+  content: VNode;
+}
+
+interface State {
+  shown: boolean;
+}
+
+export default class Modal extends Component<Props, State> {
+  private dialogElement?: HTMLDialogElement;
+
+  componentDidMount() {
+    if (!this.dialogElement) throw new Error('Modal missing');
+
+    // Set inert by default
+    this.dialogElement.inert = true;
+
+    // Prevent events from leaking through the dialog
+    this.dialogElement.onclick = (event) => {
+      event.preventDefault();
+      event.stopImmediatePropagation();
+    };
+  }
+
+  private _closeOnTransitionEnd = () => {
+    // If modal does not exist
+    if (!this.dialogElement) return;
+
+    this.dialogElement.close();
+    this.dialogElement.inert = true;
+    this.setState({ shown: false });
+  };
+
+  showModal() {
+    if (!this.dialogElement || this.dialogElement.open)
+      throw Error('Modal missing / already shown');
```

Comment:
My preference is that, if the thing is already in the desired state, then just return rather than throwing. Also, try to avoid using both component state & dialog element state. Use one as the source of truth (the component state?)